robot/web/web_c.go

```diff
@@ -375,14 +417,55 @@ func (svc *webService) handleVisualizeResourceGraph(w http.ResponseWriter, r *ht
 		}
 	}()
 
-	graph, err := graphviz.ParseBytes([]byte(dot))
+	graph, err := graphviz.ParseBytes([]byte(snapshot.Dot))
 	if err != nil {
 		return
 	}
 	if layout != "" {
 		gv.SetLayout(graphviz.Layout(layout))
 	}
-	if err = gv.Render(graph, graphviz.SVG, w); err != nil {
+
+	navButton := func(index int, label string) {
+		url := *r.URL
+		q := r.URL.Query()
+		q.Set(lookupParam, strconv.Itoa(index))
+		url.RawQuery = q.Encode()
+		var html string
+		if index < 0 || index >= snapshot.Count || index == lookup {
+			html = fmt.Sprintf(`<a>%s</a>`, label)
+		} else {
+			html = fmt.Sprintf(`<a href=%q>%s</a>`, url.String(), label)
+		}
+		write(html)
+	}
+
+	write(`<html><div>`)
+	navButton(0, "Latest")
+	write(`|`)
+	navButton(snapshot.Index-1, "Later")
+	// Index counts from 0, but we want to show pages starting from 1
+	write(fmt.Sprintf(`| %d / %d |`, snapshot.Index+1, snapshot.Count))
+	navButton(snapshot.Index+1, "Earlier")
+	write(`|`)
+	navButton(snapshot.Count-1, "Earliest")
+	write(`</div>`)
+
+	fxml := filterXML{w: w}
+	if err = gv.Render(graph, graphviz.SVG, fxml); err != nil {
 		return
 	}
+	write(`</html>`)
+}
+
+// custom writer to skip first 5 lines of xml.
```

Comment:
Assuming the slicing on 468 is the intention, 5 -> 6

Reply:
yup :(